### PR TITLE
WG: Send disconnect and create new config on connect

### PR DESCRIPTION
**Pull request by fantostisch**
This uses a modified version of eduVPN with WireGuard support. The used API can be found in [wireguard-vpn-user-portal](https://github.com/fantostisch/wireguard-vpn-user-portal/blob/v2/src/WireGuard/WireGuardApiModule.php).

### CODE REVIEW

File: app/src/main/java/nl/eduvpn/app/entity/VpnProtocol.kt [hunk 1]
```diff
@@ -1,9 +1,7 @@
 package nl.eduvpn.app.entity
 
-import com.wireguard.config.Config
-
 sealed class VpnProtocol
```
Comment (dzolnai) [inline app/src/main/java/nl/eduvpn/app/entity/VpnProtocol.kt:3]: I think you can rewrite this like:

```
sealed class VpnProtocol {
  data class OpenVPN(val profile: Profile) : VpnProtocol()
  object WireGuard : VpnProtocol()
}
```
And later you can refer to this like:
VpnProtocol.OpenVpn(profile) or VpnProtocol.WireGuard

WireGuard.INSTANCE looks a bit weird to me

**Review by dzolnai — CHANGES_REQUESTED** [review]
Almost good, just one remark.

File: app/src/main/java/nl/eduvpn/app/entity/VpnProtocol.kt (hunk 1 above)
Comment (fantostisch) [reply app/src/main/java/nl/eduvpn/app/entity/VpnProtocol.kt]: > WireGuard.INSTANCE looks a bit weird to me

I was curious how this worked, so I compiled the apk and then decompiled the apk using dex2jar and jadx to retrieve the Java code of the WireGuard class:

```java
public static final class WireGuard extends VpnProtocol {
    public static final WireGuard INSTANCE = new WireGuard();

    private WireGuard() {
        super(null);
    }
}
```

Using WireGaurd.INSTANCE seems to be the only way to get access to the WireGuard object from Java.

Comment (dzolnai) [reply app/src/main/java/nl/eduvpn/app/entity/VpnProtocol.kt]: Ah yes that is still Java, did not see that. Then it is OK as it is.

**Review by dzolnai — APPROVED** [review]

**Pull request merged**